flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -207,6 +239,9 @@ public void register(Metric metric, String metricName, MetricGroup group) {
 					}
 				}
 			}
+			if (queryService != null) {
+				notifyOfAddedMetric(queryService, metric, metricName, (AbstractMetricGroup) group);
```

Comment:
I think it's always good practice to add the class whose static method on calls in front of the static method. Otherwise you easily think that this method is somewhere defined in this class. Consequently, I would discourage static imports a little bit.